client/src/services/languages.js

```diff
@@ -21,29 +21,23 @@ const getLanguages = async (dispatch) => {
 };
 
 const updateLanguages = async (languages, dispatch) => {
-  dispatch({ type: "UPDATE_USER_LANGUAGES" });
-
-  //Make put call to api for updating languages on current user
-  const config = {
-    method: "put",
-    url: "/api/user/languages",
-    data: {
-      languages,
-    },
-  };
-
-  const response = await request.putToAPI(config);
-
-  if (typeof response.status != "undefined") {
-    console.log(response.status);
+  try {
+    const config = {
+      method: "put",
+      url: "/api/user/languages",
+      data: {
+        languages,
+      },
+    };
+
+    const response = await request.putToAPI(config);
+    dispatch({
+      type: "UPDATE_USER_LANGUAGES_SUCCESS",
+      user: response.data,
+    });
+  } catch (e) {
+    dispatch({ type: "UPDATE_USER_LANGUAGES_ERROR", error: { ...e } });
```

Comment:
A lot of the updated services file do not have that initial dispatch to change loading state. I think that should be added to any service that is making an API call.

Reply:
In Login and Signup there is no need to change state to 'loading: true' , bc if we log in or signup succesfully loading will be changed to true with getUser service and before that we can return errors from server in realtime

Reply:
got it!

Reply:
Added initial dispatch in language.js service